test/0dt/mzcompose.py

```diff
@@ -865,6 +863,343 @@ def workflow_basic(c: Composition) -> None:
+        ALTER SYSTEM SET cluster = cluster;
+    """,
+        service="mz_old",
+        port=6877,
+        user="mz_system",
+    )
+
+    inserts = (
+        "INSERT INTO postgres_source_table VALUES "
+        + ", ".join([f"({i})" for i in range(count)])
+        + ";"
+    )
+
+    start_time = time.time()
+    c.testdrive(
+        dedent(
+            f"""
+        > SET CLUSTER = cluster;
+
+        $ postgres-execute connection=postgres://postgres:postgres@postgres
+        CREATE USER postgres1 WITH SUPERUSER PASSWORD 'postgres';
+        ALTER USER postgres1 WITH replication;
+        DROP PUBLICATION IF EXISTS postgres_source;
+        DROP TABLE IF EXISTS postgres_source_table;
+        CREATE TABLE postgres_source_table (f1 INTEGER);
+        ALTER TABLE postgres_source_table REPLICA IDENTITY FULL;
+        {inserts}
+        CREATE PUBLICATION postgres_source FOR ALL TABLES;
+
+        > CREATE SECRET pgpass AS 'postgres';
+        > CREATE CONNECTION pg FOR POSTGRES
+          HOST 'postgres',
+          DATABASE postgres,
+          USER postgres1,
+          PASSWORD SECRET pgpass;
+        > CREATE SOURCE postgres_source
+          IN CLUSTER cluster
+          FROM POSTGRES CONNECTION pg
+          (PUBLICATION 'postgres_source');
+        > CREATE TABLE postgres_source_table FROM SOURCE postgres_source (REFERENCE postgres_source_table)
+        # TODO: This should also work with a default index
+        # > CREATE DEFAULT INDEX ON postgres_source_table
+        > SELECT COUNT(*) FROM postgres_source_table;
+        {count}
+        """
+        ),
+        quiet=True,
+    )
+
+    for i in range(1, repeats):
+        c.testdrive(
+            dedent(
+                f"""
+        $ postgres-execute connection=postgres://postgres:postgres@postgres
+        {inserts}
+        > SELECT COUNT(*) FROM postgres_source_table
+        {count*(i+1)}
+        """
+            ),
+            quiet=True,
+        )
+
+    elapsed = time.time() - start_time
+    print(f"initial ingestion took {elapsed} seconds")
+
+    with c.override(
+        Materialized(
+            name="mz_new",
+            sanity_restart=False,
+            deploy_generation=1,
+            system_parameter_defaults=SYSTEM_PARAMETER_DEFAULTS,
+            restart="on-failure",
+            external_metadata_store=True,
+            # environment_extra=["FAILPOINTS=hydration=sleep(12000)"],
+        )
+    ):
+        c.up("mz_new")
+        start_time = time.time()
+        c.await_mz_deployment_status(DeploymentStatus.READY_TO_PROMOTE, "mz_new")
+        elapsed = time.time() - start_time
+        print(f"re-hydration took {elapsed} seconds")
+        c.promote_mz("mz_new")
+        start_time = time.time()
+        c.await_mz_deployment_status(
+            DeploymentStatus.IS_LEADER, "mz_new", sleep_time=None
+        )
+        elapsed = time.time() - start_time
+        print(f"promotion took {elapsed} seconds")
```

Comment:
Also I notice you don't have the 5s sleep here!

Reply:
I have removed it in Kafka too now